### ISSUE TITLE
Launcher takes several seconds to fully render, about 50% of the time

### ISSUE THREAD

**Issue by spencerflagg** [nivisi/LeafyLauncher#212]
When returning to the launcher from another app, almost half the time the phone will hang on either a solid black or solid white screen for several seconds before displaying the launcher. The issue seems happen regardless of how I return to the launcher.

I've stopped using the launcher for now, as it became a significant inconvenience - but let me know if you want me to test it under certain conditions or send logs, etc.

Pixel 3 XL, Android 12
(note: the phone is several years old, with several hundred apps installed, but it is still running about as fast as ever. I almost never see lagging like this)

**Comment by nivisi** [comment]
@spencerflagg yep, it happens to me as well. I suppose it is related to our architecture that is going to be rewritten. Thanks for the report!

**Comment by nivisi** [comment]
I will actually prioritise this issue, seems like a really not cool thing.

**Comment by nivisi** [comment]
@spencerflagg hi! JFYI, this should be fixed when #213 is done. It will be a major release that will heavily clean the codebase and increase the performance (the latter — hopefully 😀). I'm also keeping in mind the things you suggested in #211, so I will probably bring those in the new release/at least will build a basement for it so you could fork and customise it as you wish.

Cheers x
